Bug: Nil Pointer Dereference in validateClustersOnUpdate

### 📜 Description

`validateClustersOnUpdate` dereferences `cluster.Status.ClusterHealth` without a nil guard. When a newly added cluster has not yet reported health status, `ClusterHealth` is `nil` and accessing `.ClusterHealthStatus` on it causes a nil pointer dereference panic.

**File:** `service/slice_config_webhook_validation.go` 


### 👟 Reproduction steps

1. Register a new cluster that has not yet reported its health status (`ClusterHealth` is `nil`).
2. Create or update a `SliceConfig` that adds this cluster to `spec.clusters`.
3. The admission webhook calls `validateClustersOnUpdate`.
4. The controller panics of `slice_config_webhook_validation.go`.


### 👍 Expected behavior

It should validate gracefully - either skip the health check when `ClusterHealth` is `nil` , or return a clear validation error.


### 👎 Actual Behavior

It panics with a nil pointer dereference, crashing the webhook admission handler and blocking all `SliceConfig` updates.


### 🐚 Relevant log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
goroutine [running]:
github.com/kubeslice/kubeslice-controller/service.validateClustersOnUpdate(...)
    service/slice_config_webhook_validation.go
```

### Version

_No response_

### 🖥️ What operating system are you seeing the problem on?

_No response_

### ✅ Proposed Solution

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [x] I checked and didn't find any similar issue

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/service/slice_config_webhook_validation.go
+++ b/service/slice_config_webhook_validation.go
@@ -347,7 +347,7 @@ func validateClustersOnUpdate(ctx context.Context, sliceConfig *controllerv1alph
 		if cluster.Status.RegistrationStatus != controllerv1alpha1.RegistrationStatusRegistered {
 			return field.Invalid(field.NewPath("Spec").Child("Clusters"), clusterName, "cluster registration is not completed. Possible cause: Slice Operator installation is pending on the cluster.")
 		}
-		if cluster.Status.ClusterHealth.ClusterHealthStatus != controllerv1alpha1.ClusterHealthStatusNormal {
+		if cluster.Status.ClusterHealth != nil && cluster.Status.ClusterHealth.ClusterHealthStatus != controllerv1alpha1.ClusterHealthStatusNormal {
 			return field.Invalid(field.NewPath("Spec").Child("Clusters"), clusterName, "cluster health is not normal")
 		}
 	}

--- a/service/slice_config_webhook_validation_test.go
+++ b/service/slice_config_webhook_validation_test.go
@@ -119,6 +119,8 @@ var SliceConfigWebhookValidationTestBed = map[string]func(*testing.T){
 	"TestValidateRotationInterval_NoChange":                                                                                    TestValidateRotationInterval_NoChange,
 	"SliceConfigWebhookValidation_UpdateValidateSliceConfigUpdatingVPNCipher":                                                  UpdateValidateSliceConfigUpdatingVPNCipher,
 	"Test_validateSlicegatewayServiceType":                                                                                     test_validateSlicegatewayServiceType,
+	"SliceConfigWebhookValidation_UpdateValidateSliceConfigWithNilClusterHealthIsAllowed":                                       TestValidateClustersOnUpdate_NilClusterHealthIsAllowed,
+	"SliceConfigWebhookValidation_UpdateValidateSliceConfigWithNilClusterHealthUnhealthy":                                       TestValidateClustersOnUpdate_NonNilClusterHealthUnhealthyIsRejected,
 }
 
 func test_validateSlicegatewayServiceType(t *testing.T) {
@@ -2299,6 +2301,62 @@ func TestValidateClustersOnUpdate_NetworkSliceOnNoNetClustersError(t *testing.T)
 	require.Contains(t, err.Error(), "cluster network is not present")
 	clientMock.AssertExpectations(t)
 
+}
+
+// TestValidateClustersOnUpdate_NilClusterHealthIsAllowed verifies that a newly-registered
+// cluster whose ClusterHealth has not yet been reported (nil pointer) does NOT cause a panic
+// and is treated as healthy (issue #315).
+func TestValidateClustersOnUpdate_NilClusterHealthIsAllowed(t *testing.T) {
+	oldSliceConfig := &controllerv1alpha1.SliceConfig{}
+	oldSliceConfig.Spec.OverlayNetworkDeploymentMode = controllerv1alpha1.NONET
+	clientMock, newSliceConfig, ctx := setupSliceConfigWebhookValidationTest("slice_config", "random")
+	newSliceConfig.Spec.OverlayNetworkDeploymentMode = controllerv1alpha1.NONET
+	// cluster-1 is a newly added cluster (not present in oldSliceConfig)
+	newSliceConfig.Spec.Clusters = []string{"cluster-1"}
+
+	clientMock.On("Get", ctx, client.ObjectKey{
+		Name:      newSliceConfig.Spec.Clusters[0],
+		Namespace: "random",
+	}, &controllerv1alpha1.Cluster{}).Return(nil).Run(func(args mock.Arguments) {
+		arg := args.Get(2).(*controllerv1alpha1.Cluster)
+		arg.Status.NodeIPs = []string{"1.1.1.1"}
+		arg.Status.RegistrationStatus = controllerv1alpha1.RegistrationStatusRegistered
+		// ClusterHealth is intentionally left nil to simulate a newly-registered cluster
+		arg.Status.ClusterHealth = nil
+	}).Twice()
+
+	// Must not panic; nil ClusterHealth should be treated as healthy
+	err := validateClustersOnUpdate(ctx, newSliceConfig, oldSliceConfig)
+	require.Nil(t, err)
+	clientMock.AssertExpectations(t)
+}
+
+// TestValidateClustersOnUpdate_NonNilClusterHealthUnhealthyIsRejected ensures that a cluster
+// with an explicitly non-normal health status is still rejected after the nil-guard is in place.
+func TestValidateClustersOnUpdate_NonNilClusterHealthUnhealthyIsRejected(t *testing.T) {
+	oldSliceConfig := &controllerv1alpha1.SliceConfig{}
+	oldSliceConfig.Spec.OverlayNetworkDeploymentMode = controllerv1alpha1.NONET
+	clientMock, newSliceConfig, ctx := setupSliceConfigWebhookValidationTest("slice_config", "random")
+	newSliceConfig.Spec.OverlayNetworkDeploymentMode = controllerv1alpha1.NONET
+	// cluster-1 is a newly added cluster (not present in oldSliceConfig)
+	newSliceConfig.Spec.Clusters = []string{"cluster-1"}
+
+	clientMock.On("Get", ctx, client.ObjectKey{
+		Name:      newSliceConfig.Spec.Clusters[0],
+		Namespace: "random",
+	}, &controllerv1alpha1.Cluster{}).Return(nil).Run(func(args mock.Arguments) {
+		arg := args.Get(2).(*controllerv1alpha1.Cluster)
+		arg.Status.NodeIPs = []string{"1.1.1.1"}
+		arg.Status.RegistrationStatus = controllerv1alpha1.RegistrationStatusRegistered
+		arg.Status.ClusterHealth = &controllerv1alpha1.ClusterHealth{
+			ClusterHealthStatus: controllerv1alpha1.ClusterHealthStatusWarning,
+		}
+	}).Once()
+
+	err := validateClustersOnUpdate(ctx, newSliceConfig, oldSliceConfig)
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "cluster health is not normal")
+	clientMock.AssertExpectations(t)
 }
 
 func setupSliceConfigWebhookValidationTest(name string, namespace string) (*utilMock.Client, *controllerv1alpha1.SliceConfig, context.Context) {

